tests/gui/test_orb.py

```diff
@@ -0,0 +1,208 @@
+    rom_interface.write_state(State(False, True, True))
+    assert False is rom_interface.read_state().move_touch_to_timer
+
+
+def test_rom_interface_write_state_move_touch_to_timer_true():
+    rom = create_move_touch_to_timer_rom(bytearray([0x8d, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_touch_to_timer
+    rom_interface.write_state(State(True, False, False))
+    assert True is rom_interface.read_state().move_touch_to_timer
+
+
+def test_rom_interface_write_state_move_touch_to_timer_invalid():
+    rom = create_move_touch_to_timer_rom(bytearray([0x8d, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_touch_to_timer
+    rom_interface.write_state(State(None, False, False))
+    assert False is rom_interface.read_state().move_touch_to_timer
+
+
+def create_move_timer_to_exit_rom(data: bytearray) -> Rom:
+    rom = Rom(bytearray([0] * 0x50000))
+    rom.write(0x68FD - 4, bytearray([0x18, 0x05, 0xF0, 0x12]))
+    rom.write(0x68FD, data)
+    rom.write(0x68FD + len(data), bytearray([0x88, 0xd0, 0x0b, 0x8c]))
+    return rom
+
+
+def test_rom_interface_read_state_move_timer_to_exit_false():
+    rom = create_move_timer_to_exit_rom(bytearray([0x8c, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_timer_to_exit
+
+
+def test_rom_interface_read_state_move_timer_to_exit_true():
+    rom = create_move_timer_to_exit_rom(bytearray([0xEA, 0xEA, 0xEA]))
+    rom_interface = RomInterface(rom)
+    assert True is rom_interface.read_state().move_timer_to_exit
+
+
+def test_rom_interface_read_state_move_timer_to_exit_invalid():
+    rom = create_move_timer_to_exit_rom(bytearray([0x11, 0x22, 0x33]))
+    rom_interface = RomInterface(rom)
+    assert None is rom_interface.read_state().move_timer_to_exit
+
+
+def test_rom_interface_write_state_move_timer_to_exit_false():
+    rom = create_move_timer_to_exit_rom(bytearray([0xEA, 0xEA, 0xEA]))
+    rom_interface = RomInterface(rom)
+    assert True is rom_interface.read_state().move_timer_to_exit
+    rom_interface.write_state(State(True, False, True))
+    assert False is rom_interface.read_state().move_timer_to_exit
+
+
+def test_rom_interface_write_state_move_timer_to_exit_true():
+    rom = create_move_timer_to_exit_rom(bytearray([0x8c, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_timer_to_exit
+    rom_interface.write_state(State(False, True, False))
+    assert True is rom_interface.read_state().move_timer_to_exit
+
+
+def test_rom_interface_write_state_move_timer_to_exit_invalid():
+    rom = create_move_timer_to_exit_rom(bytearray([0x8c, 0xf4, 0x7c]))
+    rom_interface = RomInterface(rom)
+    assert False is rom_interface.read_state().move_timer_to_exit
+    rom_interface.write_state(State(False, None, False))
```

Comment:
Either State should use the Optional keyword or None should not be considered as a valid state.